File: lager/popsong/input_format_validators/validator/validator.cpp
#include "validator.h"

void run() {
	int len = 0;
	char c;
	while ((c = Char()) != '\n') {
		assert(c == '_' || ('a' <= c && c <= 'z'));
		len++;
	}
	int maxlen = Arg("n");
	assert(2 <= len && len <= maxlen);
}

File: skolkval/testdata_tools/examples/ninetynine/input_format_validators/validator.py
import sys

def cmdlinearg(name, default=None):
    for arg in sys.argv:
        if arg.startswith(name + "="):
            return arg.split("=")[1]
    assert default is not None, name
    return default

need_mode = cmdlinearg('mode')
have_mode, seed = raw_input().split()
assert 0 <= int(seed)
assert have_mode == need_mode
sys.exit(42)
